asyncpgsa/connection.py

```diff
@@ -15,32 +15,41 @@
 _dialect.paramstyle = 'named'
 
 
+def replace_keys(keys, querystring, params, inline=False):
+    new_query = querystring
+    new_params = []
+    for key, value in enumerate(keys):
+        if inline:
+            new_query = new_query.replace(':' + value, params[key][1], 1)
+        else:
+            new_query = new_query.replace(':' + value, '$' + str(key + 1), 1)
+            new_params.append(params[key][1])
+
+    return new_query, new_params
+
+
+def get_keys(compiled):
+    import re
+    pattern = ':(\w+)(?:\W|)'
+    p = re.compile(pattern, re.M)
+    keys = re.findall(p, compiled.string)
+    params = []
+    for i in keys:
+        params.append((i, compiled.params[i]))
+    return keys, params
```

Comment:
Params includes the key, so I dont think we need to return keys and params. Just params.